text/3389-manifest-lint.md

```diff
@@ -0,0 +1,398 @@
+example, if we merged `cargo semver-check`, there are proposed lint groups
+based on what a lint's user-overideable semver-level is which couldn't be
+guarenteed to meet these requirements.  On the implementation side, this will
+require a new channel to communicate these lints to the tools (unless we infer
+order for flags/config as well) and require them to organize their data in a
+way for it to inferred.
+
+We could use an array instead of a table:
+
+Unconfigurable:
+```toml
+[lints]
+clippy = [
+  { all = "allow" },
+  { doc_markdown = "warn" },
+]
+```
+
+Configurable:
+```toml
+[[lints.clippy.all]]
+level = "allow"
+[[lints.clippy.doc_markdown]]
+level = "warn"
+```
+Where the order is based on how to pass them on the command-line.
+
+Complex TOML arrays tend to be less friendly to work with including the fact
+that TOML 1.0 does not allow multi-line inline tables.
+
+# Prior art
+[prior-art]: #prior-art
+
+Rust
+- [cargo cranky](https://github.com/ericseppanen/cargo-cranky)
+
+Python
+- [flake8](https://flake8.pycqa.org/en/latest/user/configuration.html)
+  - Format is `level = [lint, ...]`
+- [pylint](https://github.com/PyCQA/pylint/blob/main/examples/pylintrc#L402)
+  - Format is `level = [lint, ...]` but the [config file is a reflection of the CLI](https://pylint.pycqa.org/en/latest/user_guide/configuration/index.html)
+- [ruff](https://beta.ruff.rs/docs/configuration/)
+  - Format is `level = [lint, ...]`, likely due to past precedence in ecosystem (see above)
+
+Javascript
+- [eslint](https://eslint.org/docs/latest/use/configure/rules)
+  - Format is `lint = level` / `lint = [ level, additional config ]`
+
+Go
+- [golangci-lint](https://golangci-lint.run/usage/configuration/)
+  - Format is `level = [lint, ...]`
+
+Ruby
+- [rubocop](https://docs.rubocop.org/rubocop/1.45/configuration.html)
+  - Format is `Lint: Enabled: true`
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+## rustc reporting `Cargo.toml` as lint-level source
+
+Currently Rust tells you where a lint level was enabled when it emits a lint.
+`rustc` only sees that these lints are coming in from the command-line and
+doesn't know about `[lints]`.
+It would be nice if it could also point to Cargo.toml for this.  This could be
+as simple as a `--lint-source=Cargo.toml` with rustc knowing just enough about
+the `[lints]` table to process it directly.
+
+## External file
+
+Like with `package.license`, users might want to refer to an external file for
+their lints.  This especially becomes useful for copy/pasting lints between
+projects.
+
+## Configurable lints
+
+We can extend basic lint syntax:
+```toml
+[lints.clippy]
+cyclomatic_complexity = "allow"
+```
+to support configuration, whether for cargo or the lint tool:
+```toml
+[lints.clippy]
+cyclomatic_complexity = { level = "allow", rust-version = "1.23.0", threshold = 30 }
```

Comment:
imo it is a future consideration as it is an input on the schema design.  We shouldn't design assuming its a for-sure thing but if things are weighed out equally, we should probably keep the schema open to supporting it.

And yes, I understand clippy lints do not have configuration.  At the time I did not know that but I still think its useful to be concrete in examples.  I've added 2f5f8738 to try to avoid people getting the wrong idea.